service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/AllergyStructureMapper.java

```diff
@@ -262,4 +259,26 @@ private String buildNotePertinentInformation(AllergyIntolerance allergyIntoleran
             .map(Annotation::getText)
             .collect(Collectors.joining(StringUtils.SPACE));
     }
+
+    private void buildAuthor(AllergyIntolerance allergyIntolerance, AllergyStructureTemplateParametersBuilder templateParameter) {
+        if (invalidAsserter(allergyIntolerance)) {
+            if (allergyIntolerance.hasRecorder()) {
+                buildParticipant(allergyIntolerance.getRecorder(), ParticipantType.PERFORMER)
+                    .ifPresent(templateParameter::performer);
+            }
+        } else {
```

Comment:
I would maybe have another check here to see if the asserter exists. I know the invalidAsserter method checks it but could there be this case: there isnt an asserter present-> invalidAsserter returns true -> line 270 tries to access the asserter but it doesnt exist -> exception